src/IssueViz/Selection/AnalysisIssueSelectionService.cs

```diff
@@ -29,46 +29,34 @@ namespace SonarLint.VisualStudio.IssueVisualization.Selection
     [PartCreationPolicy(CreationPolicy.Shared)]
     internal sealed class AnalysisIssueSelectionService : IAnalysisIssueSelectionService
     {
-        private IAnalysisIssueVisualization selectedIssue;
-        private IAnalysisIssueFlowVisualization selectedFlow;
-        private IAnalysisIssueLocationVisualization selectedLocation;
+        public IAnalysisIssueVisualization SelectedIssue { get; private set; }
+        public IAnalysisIssueFlowVisualization SelectedFlow { get; private set; }
+        public IAnalysisIssueLocationVisualization SelectedLocation { get; private set; }
 
         public event EventHandler<SelectionChangedEventArgs> SelectionChanged;
 
-        public IAnalysisIssueVisualization SelectedIssue
+        public void Select(IAnalysisIssueVisualization issueVisualization)
         {
-            get => selectedIssue;
-            set
-            {
-                selectedIssue = value;
-                selectedFlow = GetFirstFlowOrDefault();
-                selectedLocation = GetFirstLocationOrDefault();
+            SelectedIssue = issueVisualization;
+            SelectedFlow = GetFirstFlowOrDefault();
+            SelectedLocation = GetFirstLocationOrDefault();
 
-                RaiseSelectionChanged(SelectionChangeLevel.Issue);
-            }
+            RaiseSelectionChanged(SelectionChangeLevel.Issue);
         }
 
-        public IAnalysisIssueFlowVisualization SelectedFlow
+        public void Select(IAnalysisIssueFlowVisualization flowVisualization)
         {
-            get => selectedFlow;
-            set
-            {
-                selectedFlow = value;
-                selectedLocation = GetFirstLocationOrDefault();
+            SelectedFlow = flowVisualization;
+            SelectedLocation = GetFirstLocationOrDefault();
 
-                RaiseSelectionChanged(SelectionChangeLevel.Flow);
-            }
+            RaiseSelectionChanged(SelectionChangeLevel.Flow);
         }
 
-        public IAnalysisIssueLocationVisualization SelectedLocation
+        public void Select(IAnalysisIssueLocationVisualization locationVisualization)
```

Comment:
One problem with both the old and new versions is that there's no checking of the logical consistency of the data i.e. the caller could set the flow/location to objects that aren't related to the current issue.
It's not necessarily a blocker to using this approach, but ideally it wouldn't be possible.